.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
```

Comment:
This workflow triggers on both "create PR" and "merge PR", maybe it's more efficient to split that, say, only lint on "create PR" and do SPM build on merge?

Reply:
To clarify, you mean because it triggers off of pull requests and changes to the master branch? The `pull_request` trigger uses the `opened`, `synchronize`, and `reopened` events by default, so it will only get a second set of builds if it's merged into master (as opposed to a feature/release branch).

I think there's tangible benefits having CI builds for both PRs and changes on master. A change could break SPM or CocoaPods individually, so we need to test both to avoid regressions, and if a PR hadn't been rebased on the latest commit, merging it could end up breaking master.

I also don't think there's much benefit to reducing the frequency of CI builds right now. These builds are quite fast (all four builds were complete in under 4 minutes), so they shouldn't cause any issues. If we end up seeing builds queuing up, we can look into cutting back the number of builds, but I don't think we should optimize for that unless it becomes an issue.